pkg/server/forward.go

```diff
@@ -51,68 +53,74 @@ func init() {
 	prometheus.MustRegister(overwrittenTimestamps)
 }
 
-type Store struct {
-	next   store.Store
-	url    *url.URL
-	client *http.Client
-	logger log.Logger
-}
+// ForwardHandler gets a request containing metric families and
+// converts it to a remote write request forwarding it to the upstream at fowardURL.
+func ForwardHandler(logger log.Logger, forwardURL *url.URL) http.HandlerFunc {
+	client := http.Client{Timeout: 10 * time.Second}
```

Comment:
This timeout is 10 seconds but the context for the forward request times out in 5 seconds. Why are they different? and why do we need to timeout mechanisms? I think this will lead to confusion about which one really applies. I would say let's keep just one of the timeout mechanisms.

Reply:
You're right. I added this as another safeguard, but we don't really need it here. Removing.